core/src/main/java/org/bitcoinj/net/discovery/DnsDiscovery.java

```diff
@@ -56,7 +56,7 @@ public DnsDiscovery(String[] dnsSeeds, NetworkParameters params) {
     }
 
     private static List<PeerDiscovery> buildDiscoveries(NetworkParameters params, String[] seeds) {
-        List<PeerDiscovery> discoveries = new ArrayList<PeerDiscovery>(seeds.length);
+        List<PeerDiscovery> discoveries = new ArrayList<PeerDiscovery>(seeds != null ? seeds.length : 0);
```

Comment:
For some reason I thought that it was both an initial and permanent capacity parameter, but I see it's only initial capacity so it wouldn't prevent other code from adding more items after all.